lam4doc/services/handlers.py

```diff
@@ -6,57 +6,84 @@
 from zipfile import ZipFile
 
 from eds4jinja2.builders.report_builder import ReportBuilder
 
-from lam4doc import config
-from lam4doc.config import LAM_LOGGER
-from lam4doc.services.helpers import FileInfo, IndexInfo
+from lam4doc.config import LAMConfig, HTML_REPORT_TYPE, PDF_REPORT_TYPE
 
-logger = logging.getLogger(LAM_LOGGER)
+config = LAMConfig()
+logger = logging.getLogger(config.LAM_LOGGER)
 
+FileInfo = namedtuple('FileInfo', 'location file_name')
+IndexInfo = namedtuple('IndexInfo', 'name, config_location')
 
-def generate_report(target_location: str, report_builder: ReportBuilder) -> Path:
+
+class ReportType(Exception):
+    """
+        An exception when no acceptable template location has been found.
+    """
+
+
+def get_report_location(extension: str) -> str:
+    if extension == HTML_REPORT_TYPE:
+        return config.LAM_HTML_REPORT_TEMPLATE_LOCATION
+    elif extension == PDF_REPORT_TYPE:
+        return config.LAM_PDF_REPORT_TEMPLATE_LOCATION
+    raise ReportType('No acceptable report template location found')
+
+
+def generate_report(location: str, report_builder: ReportBuilder) -> Path:
     """
     Handler for generating the lam report.
-    :param target_location: path to the report templates
+    :param location: path to the report templates
     :param report_builder: a report builder type service (eds4jinja2.builders.report_builder.ReportBuilder)
     :return: location of the report document
     """
-    logger.debug(f'start service for generating report with target_location: {target_location}')
+    logger.debug(f'start service for generating report with location: {location}')
 
     report_builder.make_document()
 
-    report_path = Path(str(target_location)) / report_builder.template
+    report_path = Path(str(location)) / report_builder.template
 
-    logger.debug(f'finish service for generating report with target_location: {target_location}')
+    logger.debug(f'finish service for generating report with location: {location}')
     return report_path
 
 
-def generate_lam_report(target_location: str) -> Path:
+def generate_lam_report(location: str, extension: str) -> Path:
     """
     Method for generating the lam report using the generic prepare_template and generate_report
-    :param target_location: path to the report templates
+    :param extension: definition for eds4jinja2 report output type
+    :param location: path to the report templates
     :return: path to report
     """
     logger.debug('start service for generating lam report')
 
-    report_builder = ReportBuilder(target_path=config.LAM_REPORT_TEMPLATE_LOCATION,
-                                   output_path=target_location,
-                                   additional_config={"conf": {"default_endpoint": config.LAM_FUSEKI_SERVICE}})
-    report_location = generate_report(target_location, report_builder)
+    additional_config = {
+        "conf": {
+            "default_endpoint": config.LAM_FUSEKI_REPORT_URL
```

Comment:
if the report is latex, we will have to add one more parameter (`template_syntax_flavour='latex'`)  to the config, just to be sure that the right template_syntax is used. Ask em if unclear.